Framework/API/src/Algorithm.cpp

```diff
@@ -1524,7 +1517,6 @@ void Algorithm::cancel() {
  * and check if the algorithm has requested that it be cancelled.
  */
 void Algorithm::interruption_point() {
-  Poco::FastMutex::ScopedLock _lock(m_mutex);
```

Comment:
Yes, I guess the follow-up question is why `interruption_point()` is called in `PARALLEL_CHECK_INTERUPT_REGION`.

See #15213.